### PR TITLE
Add recipe for `rl-games`

### PR DESCRIPTION
**Pull request by flferretti**
This is a recipe for `rl-games` pure-Python package.

PyPI: https://pypi.org/project/rl-games/
Repo: https://github.com/Denys88/rl_games

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by flferretti** [comment]
@conda-forge/help-python Ready for review, thanks!

The Windows action is failing because `pytorch` is not available.

File: recipes/rl-games/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    - pytorch
+    - numpy
+    - watchdog >=2.19
+  run_constrained:
+    - ale-py
+    - AutoROM
+    - brax
+    - jax
+    - mujoco-py
+    - envpool
```
Comment (traversaro) [inline recipes/rl-games/meta.yaml:44]: This is a bit strange. `run_constrained` is used to enforce constraint on packages that are not run dependencies, if there are no constraints why you want to have them as `run_constrained` ?

File: recipes/rl-games/02-avoid_poetry_solver.patch [hunk 2]
```diff
@@ -0,0 +1,2411 @@
...
+-[build-system]
+-requires = ["poetry-core>=1.0.0"]
+-build-backend = "poetry.core.masonry.api"
+-
+-[tool.poetry.extras]
+-atari = ["ale-py", "AutoROM"]
+-brax = ["brax", "jax"]
+-mujoco = ["mujoco-py"]
+-envpool = ["envpool"]
+
```
[the remark this answers is not in the document]
Comment (flferretti) [reply recipes/rl-games/02-avoid_poetry_solver.patch:2411]: Yes, it makes sense. Thanks!

File: recipes/rl-games/meta.yaml (hunk 1 above)
Comment (flferretti) [reply recipes/rl-games/meta.yaml:44]: I think it's a leftover from the first version of the recipe that took into account the constraints from the original `setup.py`. Thanks!

**Comment by traversaro** [comment]
Thanks @flferretti for working on this! For the review it would be helpful to:
* explain why you need to set `include_package_data` to false? 
* explain why you need to drop the use of poetry and delete the pyproject.toml, from https://github.com/conda-forge/dolo-feedstock/blob/ac4d17b604ed5fabdc40cac1bd6ecb982e714f73/recipe/meta.yaml#L12 it seems that you can use poetry also in conda-forge

**Comment by flferretti** [comment]
Thanks for review! 
> - explain why you need to set include_package_data to false?

I thought they were not useful for the recipe. If you look at https://github.com/Denys88/rl_games/blob/cba782ceb772795628e52a3da3d5dc8c20ecb779/setup.py#L20 you can see that `docs` and "." will be installed as a package, this will install [`rl_games/configs/*.yaml`](https://github.com/Denys88/rl_games/tree/master/rl_games/configs), which I believe it to be not useful in this case

> - explain why you need to drop the use of poetry and delete the pyproject.toml, from https://github.com/conda-forge/dolo-feedstock/blob/ac4d17b604ed5fabdc40cac1bd6ecb982e714f73/recipe/meta.yaml#L12 it seems that you can use poetry also in conda-forge

This is interesting, I was not aware of this! The dependencies were recognized correctly by conda-forge at first, but the problem is that they are super strict and some packages are not currently available here, e.g. `gym>=0.23, <0.24`, which pops out from `pip check`

**Comment by traversaro** [comment]
> > * explain why you need to drop the use of poetry and delete the pyproject.toml, from https://github.com/conda-forge/dolo-feedstock/blob/ac4d17b604ed5fabdc40cac1bd6ecb982e714f73/recipe/meta.yaml#L12 it seems that you can use poetry also in conda-forge
> 
> This is interesting, I was not aware of this! The dependencies were recognized correctly by conda-forge at first, but the problem is that they are super strict and some packages are not currently available here, e.g. `gym>=0.23, <0.24`. For this reason, I decided to drop poetry and make conda solve the environment

Perhaps we can use the `--only-root` option for handling this? See https://github.com/python-poetry/poetry/pull/5783 .

**Comment by traversaro** [comment]
> > > * explain why you need to drop the use of poetry and delete the pyproject.toml, from https://github.com/conda-forge/dolo-feedstock/blob/ac4d17b604ed5fabdc40cac1bd6ecb982e714f73/recipe/meta.yaml#L12 it seems that you can use poetry also in conda-forge
> > 
> > 
> > This is interesting, I was not aware of this! The dependencies were recognized correctly by conda-forge at first, but the problem is that they are super strict and some packages are not currently available here, e.g. `gym>=0.23, <0.24`. For this reason, I decided to drop poetry and make conda solve the environment
> 
> Perhaps we can use the `--only-root` option for handling this? See [python-poetry/poetry#5783](https://github.com/python-poetry/poetry/pull/5783) .

Sorry, the problem is a bit more complex. The problem here is that poetry/pyproject.toml and setup.py are basically inconsistent, i.e. they have a different set of dependencies and constraints. It seems that the one most recently updated is the `setup.py`, so indeed I think it make sense to just remove `poetry/pyproject.toml` and use `setup.py`.

**Comment by flferretti** [comment]
@conda-forge/help-python, @conda-forge/staged-recipes the recipe is now ready for review, thanks!

The Windows action is still failing because pytorch is not available.

**Comment by traversaro** [comment]
I can confirm all my doubts have been addressed.

**Pull request merged**